### PR TITLE
#120 register single dependency

### PR DESCRIPTION
**Pull request by hugosenari**
Add method `.set(name, factory, options)` as alias for `.register(factory, options)`

```javascript
exports.handler = laconia(kinesis(app))
  .set("orderStream", ({env}) => myOwnFactory(env.ORDER_STREAM_NAME) );
```

### CODE REVIEW

File: packages/laconia-core/test/laconia.spec.js [hunk 1]
```diff
@@ -171,6 +171,69 @@ describe("laconia", () => {
     });
   });
 
+  describe("#set", () => {
```
Comment (ceilfors) [inline packages/laconia-core/test/laconia.spec.js]: Great tests!

File: packages/laconia-core/src/laconia.js [hunk 2]
```diff
@@ -41,6 +41,14 @@ module.exports = app => {
       }
       return laconia;
     },
+    set: (name, factory, options = {}) => {
+      checkFunction("register", factory);
+      laconia.register(
+        async laconiaContext => ({ [name]: await factory(laconiaContext) }),
```
Comment (ceilfors) [inline packages/laconia-core/src/laconia.js]: I don't think we need the `async` keyword here.

File: packages/laconia-core/src/index.d.ts [hunk 3]
```diff
@@ -23,6 +23,11 @@ declare namespace laconia {
       factory: LaconiaFactory | LaconiaFactory[],
       options?: FactoryOptions
     ): LaconiaHandler;
+    set(
```
Comment (ceilfors) [inline packages/laconia-core/src/index.d.ts]: Would you mind adding a test under `laconia/packages/laconia-core/test/types/index.ts`? It's a single line that you should be adding.

File: packages/laconia-core/test/laconia.spec.js [hunk 4]
```diff
@@ -171,6 +171,69 @@ describe("laconia", () => {
...
+            foo: "bar",
+            boo: "baz"
+          })
+        );
+      });
+
+      it("should be able to add async instances by calling 'set'", async () => {
+        const app = jest.fn();
+        await laconia(app).set("foo", async lc => {
+          return Promise.resolve("bar");
```
Comment (geoffdutton) [inline packages/laconia-core/test/laconia.spec.js]: You don't need to use `Promise.resolve` when returning from an async function, fyi.

**Comment by geoffdutton** [comment]
> Thanks @hugosenari for creating this PR, this is a great addition to the framework!
> 
> Hey @geoffdutton! What's your thoughts on the naming of `set` for this one?

It seems a little generic in my opinion. Did you guys discuss using `.register`?

**Comment by geoffdutton** [comment]
What do y'all think about this? https://github.com/laconiajs/laconia/pull/284

**Comment by ceilfors** [comment]
Thanks @geoffdutton. That's a good point on set name being too generic. Users might mistakenly think that this method is setting a configuration or a property. @hugosenari did make an attempt to overload this at #279, but we didn't know a better way to overload the method and you've just shown us how!

@hugosenari Would you mind have a look into #284 and see if has covered your concern at #279?

**Comment by hugosenari** [comment]
> we should prefer user experience.

And #284 is better for user than this and #279 ;-)

**Pull request closed without merging**